tests/unit/amo/components/TestAddon.js

```diff
@@ -200,6 +200,32 @@ describe('Addon', () => {
     expect(root.find(AddonMeta).prop('addon')).toEqual(null);
   });
 
+  it('does not dispatch fetchAddon action when slug is the same', () => {
```

Comment:
As mentioned above, I think we should have a test for "same slug; different addonType".